galley/pkg/config/analysis/analyzers/virtualservice/regexes.go

```diff
@@ -85,6 +93,21 @@ func analyzeStringMatch(r *resource.Instance, sm *v1alpha3.StringMatch, ctx anal
 		return
 	}
 
-	ctx.Report(collections.IstioNetworkingV1Alpha3Virtualservices.Name(),
-		msg.NewInvalidRegexp(r, where, re, err.Error()))
+	// Get line number for different match field
+	var line int
+	switch where {
+	case "corsPolicy.allowOrigins":
+		line = util.ErrorLineForHTTPRegexAllowOrigins(p[0].(int), p[1].(int), r)
+	case "queryParams":
+		line = util.ErrorLineForHTTPRegexHeaderAndQueryParams(p[0].(int), p[1].(int), p[2].(string), where, r)
+	case "headers":
+		line = util.ErrorLineForHTTPRegexHeaderAndQueryParams(p[0].(int), p[1].(int), p[2].(string), where, r)
```

Comment:
You can combine these cases with a comma, see https://golang.org/doc/effective_go.html#switch